scripts/extractions/point_extractions/point_extractions.py

```diff
@@ -0,0 +1,281 @@
+
+
+def create_job_dataframe(
+    backend: Backend, split_jobs: List[gpd.GeoDataFrame]
+) -> pd.DataFrame:
+    """Create a dataframe from the split jobs, containg all the necessary information to run the job."""
+    columns = [
+        "backend_name",
+        "out_extension",
+        "start_date",
+        "end_date",
+        "s2_tile",
+        "geometry",
+    ]
+    rows = []
+    for job in split_jobs:
+        # Compute the average in the valid date and make a buffer of 1.5 year around
+        median_time = pd.to_datetime(job.valid_time).mean()
+        start_date = median_time - pd.Timedelta(days=275)  # A bit more than 9 months
+        end_date = median_time + pd.Timedelta(days=275)  # A bit more than 9 months
+        s2_tile = job.tile.iloc[0]
+        rows.append(
+            pd.Series(
+                dict(
+                    zip(
+                        columns,
+                        [
+                            backend.value,
+                            ".parquet",
+                            start_date.strftime("%Y-%m-%d"),
+                            end_date.strftime("%Y-%m-%d"),
+                            s2_tile,
+                            job.to_json(),
+                        ],
+                    )
+                )
+            )
+        )
+
+    return pd.DataFrame(rows)
+
+
+def create_datacube(
+    row: pd.Series,
+    connection: openeo.DataCube,
+    provider,
+    connection_provider,
+    executor_memory: str = "5G",
+    executor_memory_overhead: str = "2G",
+):
+    """Creates an OpenEO BatchJob from the given row information."""
+
+    # Load the temporal and spatial extent
+    temporal_extent = TemporalContext(row.start_date, row.end_date)
+
+    # Get the feature collection containing the geometry to the job
+    geometry = geojson.loads(row.geometry)
+    assert isinstance(geometry, geojson.FeatureCollection)
+
+    # Filter the geometry to the rows with the extract only flag
+    geometry = filter_extract_true(geometry)
+    assert len(geometry.features) > 0, "No geometries with the extract flag found"
+
+    # Backend name and fetching type
+    backend = Backend(row.backend_name)
+    backend_context = BackendContext(backend)
+
+    # TODO: Adjust this to the desired bands to download
+    bands_to_download = [
+        "S2-L2A-B04",
+        "S2-L2A-B08",
+    ]
+
+    fetch_type = FetchType.POINT
+
+    cube = raw_datacube_S2(
+        connection=connection,
+        backend_context=backend_context,
+        spatial_extent=geometry,
+        temporal_extent=temporal_extent,
+        bands=bands_to_download,
+        fetch_type=fetch_type,
+        distance_to_cloud_flag=True,
+        additional_masks_flag=False,
+        apply_mask_flag=True,
+    )
+
+    # Create monthly median composites
+    cube = median_compositing(cube=cube, period="month")
+
+    # Perform linear interpolation
+    cube = linear_interpolation(cube)
```

Comment:
This is not needed for Presto. In general, can we rely on the preprocessing code used for inference? Ideally we should have as much as possible joint code base where all these different pipelines make use of. Otherwise we have too many parallel tracks and increased risk of divergence.

Reply:
Changed the preprocessing to use worldcereal_preprocessed_inputs_gfmap (the same that's being used for inference)